frameworks/Swift/vapor/Sources/vapor-tfb-mysql/Routes.swift

```diff
@@ -1,73 +1,97 @@
-            let queries = queriesParam(for: req)
-            let ids = (1...queries).map({ _ in WorldMeta.randomId() })
-            var worlds = try ids.flatMap { try World.find($0) }
-            worlds.forEach { $0.randomNumber = WorldMeta.randomRandomNumber() }
-            worlds = try worlds.flatMap { world in
-                let modifiedWorld = world
-                try modifiedWorld.save()
-                return modifiedWorld
+        return try ids.map { id in
+                return try World.find(id, on: req)
+            }
+            .flatten(on: req)
+            .map(to: [World].self) { result in
+                return result.compactMap({ $0 })
             }
-            let updatedWorlds = try worlds.flatMap { try $0.makeJSON() }
-            return JSON(updatedWorlds)
-        }
     }
-}
 
+    // Test type 4: Fortunes
+//    let posixLocale = Locale(identifier: "en_US_POSIX")
+//    router.get("fortunes") { req -> Future<View> in
+//        return req.withNewConnection(to: DatabaseIdentifier.mysql, closure: { db -> Future<[Fortune]> in
+//                return db.query(Fortune.self).all()
+//                    .map(to: [Fortune].self) { fortunes in
+//                        var newFortunes = fortunes
+//                        let additional = Fortune(id: 0, message: "Additional fortune added at request time.")
+//                        newFortunes.insert(additional, at: 0)
+//                        newFortunes.sort(by: { lhs, rhs -> Bool in
+//                            return lhs.message.compare(rhs.message, locale: posixLocale) == .orderedAscending
+//                        })
+//                        return newFortunes
+//                    }
+//                    .flatMap(to: View.self) { fortunes in
+//                        return req.view().render("fortune", fortunes)
+//                    }
+//            })
+//    }
+
+    // Test type 5: Database updates
+    router.get("updates") { req -> Future<[World]> in
+        let queries = queriesParam(for: req)
+        let ids = (1...queries).map({ _ in WorldMeta.randomId() })
 
-/// Since Routes doesn't depend on anything
-/// to be initialized, we can conform it to EmptyInitializable
-///
-/// This will allow it to be passed by type.
-extension Routes: EmptyInitializable { }
+        return try ids.map { id in
+                return try World.find(id, on: req)
+            }
+            .flatten(on: req)
+            .map(to: [World].self) { result in
+                let worlds = result.compactMap({ $0 })
+                worlds.forEach { $0.randomNumber = WorldMeta.randomRandomNumber() }
+                return worlds
+            }
+            .flatMap(to: [World].self) { worlds in
+                return worlds
+                    .map { $0.save(on: req) }
+                    .flatten(on: req)
+            }.flatMap { _ in
+                return req.withNewConnection(to: .mysql, closure: { db -> Future<[World]> in
```

Comment:
`withNewConnection` here will result in quite bad performance. Can use `withPooledConnection` instead. Even better would be using the request as your `DatabaseConnectable`.